include/nlohmann/detail/input/binary_reader.hpp

```diff
@@ -100,9 +100,10 @@ class binary_reader
                    const cbor_tag_handler_t tag_handler = cbor_tag_handler_t::error)
     {
         sax = sax_;
+        input_format = format;
```

Comment:
for your comments on the `binary_writer` side, I have completed the suggested changes.

for your above comment on the `binary_reader` side, does this change requires to remove all `format` parameter from `sax_parse()` functions? leaving the `format` input parameter may become redundant if the input format is set to the `binary_reader` via initializer list of the constructor (I write mostly in C, so had to use google to understand your suggestions), let me know if you can clarify this a bit. thanks

Reply:
@nlohmann, I made this requested change in a test branch test, please see this commit (all tests compile and run without issues)

https://github.com/NeuroJSON/json/commit/ebed9280cee7da6b2b7533fdb8f314b69ec66c74

Please let me know if these are what you have in mind  - because `sax_parse(format, ...)` is directly used in [many places](https://github.com/nlohmann/json/blob/700b95f4473cac1d900b807541daad0ca3d98f52/test/src/unit-ubjson.cpp#L1617), so I kept the `format` parameter of `sax_parse`.

let me know if the above commit is ok, I will merge it with this PR. thanks

Reply:
Yes, exactly.

Reply:
thanks, merged to this PR.